### PR TITLE
[lldb] Adapt lldb-swift usage of types to always be owned by TypeLists

### PR DESCRIPTION
**Pull request by augusto2112**

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
Note that the first two commits in the PR are just cherry-picks from upstream, everything needs to be merged together though or lldb won't build.

File: lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp [hunk 1]
```diff
@@ -1310,9 +1307,9 @@ TypeSP DWARFASTParserClang::ParseSubroutine(const DWARFDIE &die,
       }
     }
   }
-  return std::make_shared<Type>(
-      die.GetID(), dwarf, attrs.name, llvm::None, nullptr, LLDB_INVALID_UID,
-      Type::eEncodingIsUID, &attrs.decl, clang_type, Type::ResolveState::Full);
+  return dwarf->MakeType(die.GetID(), attrs.name, llvm::None, nullptr,
```
Comment (adrian-prantl) [inline lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp:1310]: Are these all swift-lldb specific patches, or are these changes also upstream?

Comment (augusto2112) [reply lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp:1310]: The first two commits are upstream (they're cherry-picks), the last one is lldb-swift specific.

Comment (augusto2112) [reply lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp:1310]: I had to put them in the same PR or lldb won't build.

Comment (adrian-prantl) [reply lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp:1310]: Sorry — I keep forgetting that you *can* look at individual commits' contributions in github's web interface.

**Review by adrian-prantl — APPROVED** [review]

**Pull request merged**